File: Flutter/perimeterx/ios/Runner/HumanManager.swift
import Foundation
import Flutter
import PerimeterX_SDK

class HumanManager {
    
    static let shared = HumanManager()
    
    func start() {
        do {
            let policy = HSPolicy()
            policy.automaticInterceptorPolicy.interceptorType = .none
            policy.doctorAppPolicy.enabled = true
            try HumanSecurity.start(appId: "PXj9y4Q8Em", delegate: nil, policy: policy)
        }
        catch {
            print("error: \(error)")
        }
    }
    
    func setupChannel(with controller: FlutterViewController) {
        let humanChannel = FlutterMethodChannel(name: "com.humansecurity/sdk", binaryMessenger: controller.binaryMessenger)
        humanChannel.setMethodCallHandler({(call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
            if call.method == "humanGetHeaders" {
                var json: String?
                do {
                    let headers = HumanSecurity.headersForURLRequest()
                    let data = try JSONSerialization.data(withJSONObject: headers)
                    json = String(data: data, encoding: .utf8)
                }
                catch {
                    print("error: \(error)")
                }
                result(json)
            }
            else if call.method == "humanHandleResponse" {
                if let response = call.arguments as? String, let data = response.data(using: .utf8), let httpURLResponse = HTTPURLResponse(url: URL(string: "https://example.com")!, statusCode: 403, httpVersion: nil, headerFields: nil) {
                    let handled = HumanSecurity.handleResponse(response: httpURLResponse, data: data) { challengeResult in
                        result(challengeResult == .solved ? "solved" : "cancelled")
                    }
                    if handled {
                        return
                    }
                }
                result("false")
            }
            else {
                result("")
            }
        })
    }
}

File: Flutter/perimeterx/android/app/src/main/kotlin/com/perimeterx/perimeterx/HumanManager.kt
package com.perimeterx.perimeterx

import android.app.Application
import com.perimeterx.mobile_sdk.HumanSecurity
import com.perimeterx.mobile_sdk.main.policy.HSPolicy
import com.perimeterx.mobile_sdk.main.policy.HSAutomaticInterceptorType
import com.perimeterx.mobile_sdk.HumanChallengeResult
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import org.json.JSONObject

class HumanManager {

    companion object {

        fun start(application: Application) {
            try {
                val policy = HSPolicy()
                policy.automaticInterceptorPolicy.interceptorType = HSAutomaticInterceptorType.NONE
                policy.doctorAppPolicy.enabled = true
                HumanSecurity.start(application, "PXj9y4Q8Em", null, policy)
            }
            catch (exception: Exception) {
                println("failed to start. exception: $exception")
            }
        }

        fun handleEvent(call: MethodCall, result: MethodChannel.Result) {
            if (call.method == "humanGetHeaders") {
                val json = JSONObject((HumanSecurity.headersForURLRequest(null) as Map<*, *>?))
                result.success(json.toString())
            }
            else if (call.method == "humanHandleResponse") {
                val handled = HumanSecurity.handleResponse(call.arguments!! as String, null) { challengeResult: HumanChallengeResult ->
                    result.success(if (challengeResult == HumanChallengeResult.SOLVED) "solved" else "cancelled")
                    null
                }
                if (!handled) {
                    result.success("false")
                }
            }
            else {
                result.notImplemented()
            }
        }
    }
}
